### PR TITLE
Dlg Row Stats

### PR DESCRIPTION
**Pull request by stevenndungu**
Not fully complete, one more slight thing to be checked on

### CODE REVIEW

**Comment by lilyclements** [comment]
How is this coming on? Is it ready for a review?

**Comment by stevenndungu** [comment]
You can review it, though am to rewrite its sub-dialogue

**Comment by stevenndungu** [comment]
ooh,nit really ...few changes to be made first,

**Comment by stevenndungu** [comment]
@africanmathsinitiative/instat  this is now ready for review too. thanks

File: instat/dlgRowStats.vb [hunk 1]
```diff
@@ -50,62 +59,44 @@ Public Class dlgRowStats
...
+        ucrPanelStatistics.AddRadioButton(rdoMedian, "median")
+        ucrPanelStatistics.AddRadioButton(rdoNumberofMissing, "function(z) sum(is.na(z))")
+        ucrPanelStatistics.AddRadioButton(rdoStandardDeviation, "sd")
+        ucrPanelStatistics.AddRadioButton(rdoMaximum, "max")
+        ucrPanelStatistics.AddRadioButton(rdoCount, "function(z) sum(!is.na(z))")
 
+        ucrSaveResults.SetPrefix("row_summary")
+        ucrSaveResults.SetSaveTypeAsColumn()
+        ucrSaveResults.SetDataFrameSelector(ucrSelectorForRowStats.ucrAvailableDataFrames)
+        ucrSaveResults.SetLabelText("Row Summary")
```
Comment (lilyclements) [inline instat/dlgRowStats.vb]: We want this to have a colon

File: instat/dlgRowStats.Designer.vb [hunk 2]
```diff
@@ -156,79 +194,38 @@ Partial Class dlgRowStats
...
-        '
-        'ucrBase
-        '
-        Me.ucrBase.Location = New System.Drawing.Point(10, 303)
-        Me.ucrBase.Name = "ucrBase"
-        Me.ucrBase.Size = New System.Drawing.Size(410, 52)
-        Me.ucrBase.TabIndex = 7
+        Me.ucrSaveResults.Location = New System.Drawing.Point(12, 309)
+        Me.ucrSaveResults.Name = "ucrSaveResults"
+        Me.ucrSaveResults.Size = New System.Drawing.Size(294, 24)
```
Comment (lilyclements) [inline instat/dlgRowStats.Designer.vb]: This is a little wide at the moment which means that the Row Summary and the combo box are quite far apart. Perhaps a distance of 258, 24 could work?

File: instat/dlgRowStats.vb [hunk 3]
```diff
@@ -50,62 +59,44 @@ Public Class dlgRowStats
...
 
     Private Sub ucrBase_ClickReset(sender As Object, e As EventArgs) Handles ucrBase.ClickReset
         SetDefaults()
+        SetRCodeforControls(True)
+        TestOKEnabled()
     End Sub
 
-    Private Sub ucrInputcboRowSummary_NameChanged() Handles ucrInputcboRowSummary.NameChanged
-        ucrBase.clsRsyntax.SetAssignTo(strAssignToName:=ucrInputcboRowSummary.GetText, strTempDataframe:=ucrSelectorForRowStats.ucrAvailableDataFrames.cboAvailableDataFrames.Text, strTempColumn:=ucrInputcboRowSummary.GetText)
+    Private Sub Controls_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrSelectorForRowStats.ControlContentsChanged, ucrReceiverForRowStatistics.ControlContentsChanged, ucrSaveResults.ControlContentsChanged
```
Comment (lilyclements) [inline instat/dlgRowStats.vb]: How come this has selector.controlcontentschanged when you haven't included anything about selector in the TestOKEnabled sub?

File: instat/dlgRowStats.vb (hunk 1 above)
Comment (lilyclements) [inline instat/dlgRowStats.vb]: We want to have a colon here

File: instat/dlgRowStats.Designer.vb (hunk 2 above)
Comment (lilyclements) [inline instat/dlgRowStats.Designer.vb]: Currently there is a big space between "Row Summary" and the input box, so you need to make the length a bit smaller

**Comment by lilyclements** [comment]
Something really strange has happened, I think I'll tag in @dannyparsons to this

image

I ran four different row summaries separately for two variables. The first one was fine, the second one added an empty column into the dataset and put my results in the same column as the first one.
The third one was fine, and the fourth did the same thing again

# dlgRowStats code generated by the dialog Row Statistics
row_summary1 <- apply(X=InstatDataObject$get_columns_from_data(data_name="data1", col_names="data.2", force_as_data_frame=TRUE, use_current_filter=FALSE), MARGIN=1, FUN=mean)
InstatDataObject$add_columns_to_data(col_data=row_summary1, col_name="row_summary1", data_name="data1")


# dlgRowStats code generated by the dialog Row Statistics
row_summary2 <- apply(X=InstatDataObject$get_columns_from_data(force_as_data_frame=TRUE, use_current_filter=FALSE, data_name="data1", col_names="data.2"), FUN=function(z) sum(is.na(z)), MARGIN=1)
InstatDataObject$add_columns_to_data(col_data=row_summary2, col_name="row_summary2", data_name="data1")


# dlgRowStats code generated by the dialog Row Statistics
row_summary3 <- apply(X=InstatDataObject$get_columns_from_data(data_name="data1", col_names="data.2", force_as_data_frame=TRUE, use_current_filter=FALSE), MARGIN=1, FUN=min)
InstatDataObject$add_columns_to_data(col_data=row_summary3, col_name="row_summary3", data_name="data1")

# dlgRowStats code generated by the dialog Row Statistics
row_summary4 <- apply(X=InstatDataObject$get_columns_from_data(force_as_data_frame=TRUE, use_current_filter=FALSE, data_name="data1", col_names="data.2"), FUN=sd, MARGIN=1)
InstatDataObject$add_columns_to_data(col_data=row_summary4, col_name="row_summary4", data_name="data1")

**Comment by lilyclements** [comment]
Additionally, it isn't running the correct code - the mean is not correct (the first set of values in row_summary1) etc

File: instat/dlgRowStats.vb (hunk 1 above)
Comment (stevenndungu) [inline instat/dlgRowStats.vb]: That was added on the commit above

File: instat/dlgRowStats.Designer.vb (hunk 2 above)
Comment (stevenndungu) [inline instat/dlgRowStats.Designer.vb]: Ok will make the adjustment, the control was made with that space length

**Comment by stevenndungu** [comment]
Sorry, something must have gone wrong somewhere. We can skype tomorrow on this, because on my branch everything runs well (just tested now). Thanks

**Comment by lilyclements** [comment]
I'm testing on the newest version, are you? I'm not sure how I did it, I put in the first one and made the column size wider as I wanted to see the name of the column, and then that kept happening.

**Comment by stevenndungu** [comment]
@africanmathsinitiative/developers this is ready for re-review. Required changes were made.

**Comment by maxwellfundi** [comment]
@stevenndungu  this dialog is working fine except fot one very minor thing. The Ucr save does not reset on clicking reset. 

You might want to add a line for resetting the ucrsave then this will be ready for @dannyparsons .

**Pull request merged**